### PR TITLE
Meal & Reservation endpoints

### PR DESCRIPTION
**Pull request by Atefeh-Behroozi**
Meal & Reservation endpoints homework.

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -3,13 +3,69 @@ const router = express.Router();
...
   } catch (error) {
-    throw error;
+    res.status(500).json({ error: 'An error occurred while retrieving meals' });
   }
 });
 
+// POST a new meal
+router.post('/', async (req, res) => {
+  try {
+    const meal = await knex('meals').insert(req.body);
```
Comment (raihanmehran) [inline src/backend/api/meals.js:24]: your code works fine, but it is better to validate the `req.body` before inserting it to database

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -3,13 +3,69 @@ const router = express.Router();
...
+    }
+  } catch (error) {
+    res.status(500).json({ error: 'An error occurred while retrieving the meal' });
+  }
+});
+
+// PUT update a meal by id
+router.put('/:id', async (req, res) => {
+  try {
+    const meal = await knex('meals').where({ id: req.params.id }).update(req.body);
```
Comment (raihanmehran) [inline src/backend/api/meals.js:48]: same here validate the id and body of the request

**Review by raihanmehran — COMMENTED** [review]
And better to apply the same suggestions from meals.js to reservations.js

- and one more thing, I think it is better to move the `reservations.js` file to `api` folder

**Comment by Atefeh-Behroozi** [comment]
> And better to apply the same suggestions from meals.js to reservations.js

> @raihanmehran I'm not sure I uderstand what do you mean about same suggestions? Would you please explain more?





> * and one more thing, I think it is better to move the `reservations.js` file to `api` folder

**Comment by raihanmehran** [comment]
Well, as you can see that I made some suggestions in meals.js for `id` and `req.body`. Same here in reservations.js endpoints you can apply. I hope it’s clear. 

If there is still any doubt or confusion, don’t hesitate, feel free to reach out 😊.

**Comment by Atefeh-Behroozi** [comment]
Ok, thank you for your explain. Now it is clear for me :)

**Comment by Atefeh-Behroozi** [comment]
Thank you for feedback. I validated req.body in reservation.js and meal.js files.

**Pull request merged**